server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -887,6 +881,24 @@ public Void withHandle(Handle handle) throws Exception
     return ImmutableSet.copyOf(res);
   }
 
+  /**
+   * Get the ResultSet for indexInfo for given table
+   *
+   * @param databaseMetaData DatabaseMetaData
+   * @param tableName        Name of table
+   * @return ResultSet with index info
+   */
+  public ResultSet getIndexInfo(DatabaseMetaData databaseMetaData, String tableName) throws SQLException
+  {
+    return databaseMetaData.getIndexInfo(
+        null,
+        null,
+        tableName,  // tableName is case-sensitive in mysql default setting
```

Comment:
what was the reason for wrapping this in upperCase call anyway?

Reply:
Default Derby settings needs tableName in uppercase, else it does not return the indexInfo.